apps/studio/src/components/PageEditor/ComponentSelector.tsx

```diff
@@ -104,8 +105,15 @@ function ComponentSelector() {
     setSavedPageState,
     setPreviewPageState,
   } = useEditorDrawerContext()
+  const { mutate } = trpc.page.updatePageBlob.useMutation()
+  // TODO: get this dynamically
+  const pageId = 1
+  const siteId = 1
```

Comment:
suggest to take this in as a prop also instead of retrieving from query params, since we don't want to implicitly tie this component to the page.
can see other examples of `useQueryParse` or something i forgot the name

Reply:
not very sure i understand you here - i agree on takign this as a prop but then if we do that, we shouldn't utilise `useQueryParse` right? 

i'll go with ^ and pass the props in via parent